### PR TITLE
Add some more SecKey functionality

### PR DESCRIPTION
**Pull request by steven-joruk**
There's plenty more to add relating to SecKey. This at least enables me to verify that an item in the keychain is accessible by attempting to use it to create a signature while having user interaction disabled.

I tried to make the Algorithm enum a bit less horrific by using a macro, but the approach I attempted involved concat_ident which is a nightly only feature. I'll see if there's a nicer way of doing this if you don't know of one yourself.

### CODE REVIEW

File: security-framework/src/key.rs [hunk 1]
```diff
@@ -24,6 +29,308 @@ impl_TCFType!(SecKey, SecKeyRef, SecKeyGetTypeID);
 unsafe impl Sync for SecKey {}
 unsafe impl Send for SecKey {}
 
+#[cfg(any(feature = "OSX_10_12", target_os = "ios"))]
+#[allow(missing_docs)]
+/// The algorithms that cryptographic keys enable.
+pub enum Algorithm {
```
Comment (kornelski) [inline security-framework/src/key.rs:35]: I imagine Apple will add more algorithms in the future. Can you make this enum non_exhaustive?

Comment (kornelski) [reply security-framework/src/key.rs:35]: Also there names are repeated a few times. Maybe you could use a macro to reduce the boilerplate?

```rust
macro_rules! names {
    ($($i:ident => $x:ident),*) => {
    	extern "C" {
       		$(static $x: u32;)*
    	}
        enum Enum {
            $( $i ),*
        }
        fn from(m: Enum) -> u32 {
            unsafe { match m {
                $( $i => $x, )*
            } }
        }
    }
}

names! {
    foo => macFoo, bar => macBar, baz => macBaz
}
```

File: security-framework/src/os/macos/keychain.rs [hunk 2]
```diff
@@ -74,6 +74,33 @@ impl SecKeychain {
             ))
         }
     }
+
+    #[cfg(target_os = "macos")]
+    /// Disables the user interface for keychain services functions that
+    /// automatically display a user interface.
+    pub fn disable_user_interaction() -> Result<KeychainUserInteractionLock> {
+        let code = unsafe { SecKeychainSetUserInteractionAllowed(0u8) };
```
Comment (kornelski) [inline security-framework/src/os/macos/keychain.rs:82]: why not `set_user_interaction_allowed`?

**Comment by steven-joruk** [comment]
@kornelski I'll update with your requests, cheers.

One thing I don't like is that this exposes the CFError, requiring users to add a direct dependency on core foundation. I looked at changing Error to an enum that contains either an OSStatus or CFError, but CFError is not Copy. Do you think it's a problem, and how could it be better?

**Comment by kornelski** [comment]
I don't see a problem with exposing `CFError`. There already are other methods that expose it.

**Pull request merged**